src/message/dkim.rs

```diff
@@ -0,0 +1,318 @@
+use crate::message::{header::HeaderName, Headers, Message};
+use base64::encode;
+use regex::Regex;
+use rsa::{pkcs1::FromRsaPrivateKey, Hash, PaddingScheme, RsaPrivateKey};
+use sha2::{Digest, Sha256};
+use std::fmt::Display;
+use std::time::SystemTime;
+
+/// Describe Dkim Canonicalization to apply to either body or headers
+#[derive(Copy, Clone)]
```

Comment:
Unless there is a good reason not to we should probably also derive Debug for all of these.

```suggestion
#[derive(Copy, Clone, Debug)]
```